lib/util/Components.js

```diff
@@ -419,14 +419,81 @@ function componentRule(rule, context) {
       return utils.isReturningJSX(ASTNode, strict) || utils.isReturningNull(ASTNode);
     },
 
+    getComponentNameFromJSXElement(node) {
+      if (node.type !== 'JSXElement') {
+        return null;
+      }
+      if (node.openingElement && node.openingElement.name && node.openingElement.name.name) {
+        return node.openingElement.name.name;
+      }
+      return null;
+    },
+
+    /**
+     *
+     * @param {object} node
+     * Getting the first JSX element's name.
+     */
+    getNameOfWrappedComponent(node) {
+      if (node.length && node[0].body) {
+        const body = node[0].body;
+        let componentName;
+        if (body.type === 'JSXElement') {
+          componentName = this.getComponentNameFromJSXElement(body);
+        } else if (body.type === 'BlockStatement') {
+          const innerBody = body.body;
+          innerBody.forEach(item => {
+            if (item.type === 'ReturnStatement') {
+              componentName = this.getComponentNameFromJSXElement(item.argument);
+            }
+          });
+        }
+        return componentName;
+      }
+      return null;
+    },
+
+    /**
+     * Get the list of names of components created till now
+     */
+    getDetectedComponents() {
+      const list = components.list();
+      const componentList = [];
+      Object.keys(list).forEach(key => {
+        const val = list[key];
+        if (val.node.type === 'ClassDeclaration') {
+          componentList.push(val.node.id.name);
```

Comment:
could this be a map and a `.filter(Boolean)`, rather than mutating an array inside a forEach?